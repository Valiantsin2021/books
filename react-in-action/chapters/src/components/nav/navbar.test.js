import React from "react";
import renderer from "react-test-renderer";

import Navigation from "../../../src/components/nav/navbar";

describe("Navigation", () => {
  // (9.2.5) if there is any logic in the component, cover those by varying what drives that logic (the props)
  test("snapshot, user not authenticated", () => {
    const props = {
      user: {
        name: 'username',
        authenticated: false,
        profilePicture: 'pic'
      }
    }

    const navigation = renderer.create(<Navigation {...props}/>).toJSON();
    expect(navigation).toMatchSnapshot();
  });

  test("snapshot, user is authenticated", () => {
    const props = {
      user: {
        name: 'username',
        authenticated: true,
        profilePicture: 'pic'
      }
    }

    const navigation = renderer.create(<Navigation {...props}/>).toJSON();
    expect(navigation).toMatchSnapshot();
  });
});
