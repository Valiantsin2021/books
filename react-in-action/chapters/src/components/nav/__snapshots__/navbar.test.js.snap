// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation snapshot, user is authenticated 1`] = `
<nav
  className="navbar"
>
  <div
    className="logo"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "1.75em",
      }
    }
  >
    L
  </div>
  <span
    className="user-nav-widget"
  >
    <span>
      username
    </span>
    <img
      alt="username"
      className="img-circle"
      src="pic"
      width={40}
    />
    <span
      onClick={[Function]}
    >
      <i
        className="fa fa-sign-out"
      />
    </span>
  </span>
</nav>
`;

exports[`Navigation snapshot, user not authenticated 1`] = `
<nav
  className="navbar"
>
  <div
    className="logo"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "1.75em",
      }
    }
  >
    L
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Log in or sign up
  </button>
</nav>
`;
